src/mongo/db/catalog/index_key_validate.cpp

```diff
@@ -330,6 +334,19 @@ StatusWith<BSONObj> validateIndexSpec(
             }
 
             hasIndexNameField = true;
+        } else if (IndexDescriptor::kHiddenFieldName == indexSpecElemFieldName) {
+            if (isFeatureDisabled) {
+                return {ErrorCodes::Error(31449),
+                        "Hidden indexes can only be created with FCV 4.4"};
+            }
+            if (indexSpecElem.type() != BSONType::Bool) {
+                return {ErrorCodes::TypeMismatch,
+                        str::stream()
+                            << "The field '" << IndexDescriptor::kIndexNameFieldName
+                            << "' must be a bool, but got " << typeName(indexSpecElem.type())};
+            }
+            hasHiddenField = true;
```

Comment:
I see this is a pattern in this function, but it doesn't seem to have a side-effect? Is that expected?

Reply:
Yes，that can be removed.